tests/contracts/testProvider.nim

```diff
@@ -0,0 +1,161 @@
+import pkg/chronos
+import codex/contracts
+import ../ethertest
+import ./time
+import ./helpers/mockprovider
+
+# to see supportive information in the test output
+# use `-d:"chronicles_enabled_topics:testProvider:DEBUG` option
+# when compiling the test file
+logScope:
+  topics = "testProvider"
+
+ethersuite "Provider":
```

Comment:
🟡 Could the tests that don't use `ethProvider` (using `MockProvider` instead) be in their own unit test suite? That way we could avoid the setup and teardown of `ethersuite` when not needed for those tests.